Documentation/network/clustermesh/mcsapi.rst

```diff
@@ -43,60 +47,26 @@ To enable MCS-API support on an existing Cilium installation, run:
 
 Also checkout the :ref:`EndpointSlice synchronization <endpointslicesync>` feature if you need Headless Services support.
 
-Installing CoreDNS multicluster
--------------------------------
-
-You also need to install and configure the `multicluster CoreDNS plugin`_.
-
-.. _`multicluster CoreDNS plugin`: https://github.com/coredns/multicluster
-
-The multicluster CoreDNS plugin is currently not provided by default in
-CoreDNS so you will have to recompile it yourself.
-
-To rebuild the CoreDNS image you need to first clone the `CoreDNS repo`_:
-
-   .. code-block:: shell-session
-
-      git clone https://github.com/coredns/coredns.git
-      git checkout v1.11.4
-
-
-Then you need add the multicluster plugin to the ``plugins.cfg`` file. The
-ordering of plugins matters, add it just below kubernetes plugin that has very
-similar functionality:
-
-   .. parsed-literal::
-
-      ...
-      kubernetes:kubernetes
-      multicluster:github.com/coredns/multicluster
-      ...
-
-.. _CoreDNS repo: https://github.com/coredns/coredns
-
-Then you can build your image simply by running ``make`` and then ``docker build .``
-with the right docker tag and upload it to your preferred registry.
-
-You also need to make sure that CoreDNS is able to read and watch the relevant
-MCS-API resources (ServiceImports). You can do so by running the following
-command on each cluster:
+By default Cilium will automatically configure and rollout CoreDNS for MCS-API support.
+You can also do it manually by setting the Helm value ``clustermesh.corednsMCSAPIAutoconfigure.enabled``
+to ``false`` and running:
 
    .. code-block:: shell-session
 
-      kubectl patch clusterrole system:coredns --type=json \
-         -p='[{"op":"add","path":"/rules/-","value":{"apiGroups":["multicluster.x-k8s.io"],"resources":["serviceimports"],"verbs":["list","watch"]}}]'
-
-After that you will need to update the CoreDNS's Corefile to also enable the
-multicluster plugin, for instance by executing the following command on each cluster:
-
-   .. code-block:: shell-session
+      # Adding RBAC to read SericeImports
+      kubectl create clusterrole coredns-mcsapi \
+         --verb=list,watch --resource=serviceimports.multicluster.x-k8s.io
+      kubectl create clusterrolebinding coredns-mcsapi \
+         --clusterrole=coredns-mcsapi --serviceaccount=kube-system:coredns
 
+      # Configure CoreDNS to support MCS-API
       kubectl get configmap -n kube-system coredns -o yaml | \
-         sed -E -e 's/^(\s*)kubernetes.*cluster\.local.*$/\1multicluster clusterset.local\n&/' | \
+         sed -e 's/cluster\.local/cluster.local clusterset.local/g' | \
```

Comment:
Nit. I'd use `-E` in both cases for consistency, unless there's a strong reason not to.